apps/studio/components/ui/FileExplorerAndEditor/FileExplorerAndEditor.tsx

```diff
@@ -0,0 +1,191 @@
+    isEditing?: boolean
+    originalId: number
+  }
+  children?: TreeNode[]
+}
+
+interface FileExplorerAndEditorProps {
+  files: FileData[]
+  onFilesChange: (files: FileData[]) => void
+  aiEndpoint?: string
+  aiMetadata?: {
+    projectRef?: string
+    connectionString?: string
+    includeSchemaMetadata?: boolean
+  }
+}
+
+const getLanguageFromFileName = (fileName: string): string => {
+  const extension = fileName.split('.').pop()?.toLowerCase()
+  switch (extension) {
+    case 'ts':
+    case 'tsx':
+      return 'typescript'
+    case 'js':
+    case 'jsx':
+      return 'javascript'
+    case 'json':
+      return 'json'
+    case 'html':
+      return 'html'
+    case 'css':
+      return 'css'
+    case 'md':
+      return 'markdown'
+    default:
+      return 'typescript' // Default to typescript
+  }
+}
+
+const FileExplorerAndEditor = ({
+  files,
+  onFilesChange,
+  aiEndpoint,
+  aiMetadata,
+}: FileExplorerAndEditorProps) => {
+  const selectedFile = files.find((f) => f.selected) ?? files[0]
+
+  const handleChange = (value: string) => {
+    const updatedFiles = files.map((file) =>
+      file.id === selectedFile.id ? { ...file, content: value } : file
+    )
+    onFilesChange(updatedFiles)
+  }
+
+  const addNewFile = () => {
+    const newId = Math.max(...files.map((f) => f.id)) + 1
+    const updatedFiles = files.map((f) => ({ ...f, selected: false }))
+    onFilesChange([
+      ...updatedFiles,
+      {
+        id: newId,
+        name: `file${newId}.ts`,
+        content: '',
+        selected: true,
+      },
+    ])
+  }
+
+  const handleFileNameChange = (id: number, newName: string) => {
+    if (!newName.trim()) return // Don't allow empty names
+    const updatedFiles = files.map((file) => (file.id === id ? { ...file, name: newName } : file))
+    onFilesChange(updatedFiles)
+  }
+
+  const handleDoubleClick = (id: number, currentName: string) => {
+    const newName = prompt('Enter new file name:', currentName)
+    if (newName && newName !== currentName) {
+      handleFileNameChange(id, newName)
+    }
+  }
+
+  const handleFileSelect = (id: number) => {
+    const updatedFiles = files.map((file) => ({
+      ...file,
+      selected: file.id === id,
+    }))
+    onFilesChange(updatedFiles)
+  }
+
+  const treeData = {
+    name: '',
+    children: files.map((file) => ({
+      id: file.id.toString(),
+      name: file.name,
+      metadata: {
+        isEditing: false,
+        originalId: file.id,
+      },
+    })),
+  }
+
+  return (
+    <div className="flex-1 overflow-hidden flex h-full">
+      <div className="w-64 border-r bg-surface-200 flex flex-col">
+        <div className="py-4 px-6 border-b flex items-center justify-between">
+          <h3 className="text-sm font-medium">Files</h3>
+          <Button size="tiny" type="default" icon={<Plus size={14} />} onClick={addNewFile}>
+            Add File
+          </Button>
+        </div>
+        <div className="flex-1 overflow-y-auto p-2">
```

Comment:
i reckon theres no need to add padding here? let it take the full width and height 
![image](https://github.com/user-attachments/assets/4be3e572-884a-43ee-90f9-0e69f5b684fd)